### ISSUE TITLE
How to add framework include search path?

### ISSUE THREAD

**Issue by durswd** [mono/CppSharp#589]
I try to run QtSharp on Mac.
To run it, I need to add add directory to framework include search path.
For example /Users/xxx/Qt5.5.0/5.5/clang_64/lib.

On clang, It realized by -F option, 
so currently I use addArguments "-F/Users/xxx/Qt5.5.0/5.5/clang_64/lib",

Does smart method to add framework include search path exists on CppSharp?

**Comment by ddobrev** [comment]
Hi, thanks for trying Qt#. Please bear in mind that it has not been run on OS X yet so you might run into a few issues.
Include paths are added through DriverOptions.addIncludeDir(). However, in the case of Qt#, they should be automatically detected as long as you pass the path to your qmake executable. As you can see in Program.cs in the source code, qmake is then invoked to get the include paths and all other necessary information.

**Comment by durswd** [comment]
I tried to change addArguments into addIncludeDir.
But the result became bad.
Framework include search path is not equal to include search path.
OS X framework which contains header file is included on the path.

By the way, I try to use qmake.

**Comment by ddobrev** [comment]
Could you please try DriverOptions.addSystemIncludeDir() instead?

**Comment by tritao** [comment]
@ddobrev: durswd is talking about frameworks paths, which are different from include paths.

@durswd Right now we do not have any high level option to add framework paths. The way you suggested, `addArguments("-F/Users/xxx/Qt5.5.0/5.5/clang_64/lib")` should be the best way for now.

If you guys think it's worth it, then we can add a dedicated option for it in C++#.

**Comment by ddobrev** [comment]
@tritao you're right, my mistake. I definitely think a separate option is necessary because it's a pretty basic feature. @durswd we would appreciate a patch. It should be relatively simple to do, just mirror the code we added for other similar options.
